runtime/lua/vim/semver.lua

```diff
@@ -0,0 +1,259 @@
+---@param prerelease_and_build string
+---@return boolean
+local function is_prerelease_and_build_valid(prerelease_and_build)
+  local has_build = prerelease_and_build:match('^%+[%w%.]+$')
+  local has_prerelease = prerelease_and_build:match('^%-[%w%.]+$')
+  local has_prerelease_and_build = prerelease_and_build:match('^%-[%w%.]+%+[%w%.]+')
+  return has_build or has_prerelease or has_prerelease_and_build
+end
+
+---@private
+---@param prerelease_and_build string
+---@return boolean
+local function has_prerelease_and_build(prerelease_and_build)
+  return prerelease_and_build ~= nil and prerelease_and_build ~= ''
+end
+
+---@private
+---@param version string
+---@return string
+local function create_err_msg(version)
+  return string.format('%s is not a valid version', version)
+end
+
+--- Parses a semantically formatted version string into a table.
+---
+--- Supports leading "v" and leading and trailing whitespace in the version
+--- string. e.g. `" v1.0.1-rc1+build.2"` , `"1.0.1-rc1+build.2"`, `"v1.0.1-rc1+build.2"`
+--- and `"v1.0.1-rc1+build.2 "` will be parsed as:
+---
+--- `{ major = 1, minor = 0, patch = 1, prerelease = 'rc1', build = 'build.2' }`
+---
+---@param version string Version string to be parsed.
+---@param opts table|nil Optional keyword arguments:
+---                      - strict (boolean):  when set to `true` an error will be thrown for version
+---                      strings which are not conforming to semver sepecifications (v2.0.0), (see
+---                      semver.org/spec/v2.0.0.html for details), this means that
+---                      `semver.parse('v1.2)` will throw an error. When set to `false`,
+---                      `semver.parse('v1.2)` will coerce 'v1.2' to 'v1.2.0' and return the table:
+---                      `{ major = 1, minor = 2, patch = 0 }`. Defaults to false.
+---@return table parsed_version Parsed version table
+function M.parse(version, opts)
+  if type(version) ~= 'string' then
+    error(create_err_msg(version))
+  end
+
+  opts = opts or { strict = false }
+
+  version = vim.trim(version)
+
+  local extract_components = opts.strict and extract_components_strict or extract_components_loose
+  local major, minor, patch, prerelease_and_build = extract_components(version)
+
+  -- If major is nil then that means that the version does not begin with a
+  -- digit with or without a "v" prefix.
+  if major == nil then
+    error(create_err_msg(version))
+  end
+
+  if
+    has_prerelease_and_build(prerelease_and_build)
+    and not is_prerelease_and_build_valid(prerelease_and_build)
+  then
+    error(create_err_msg(version))
+  end
+
+  local prerelease = nil
+  local build = nil
+  if prerelease_and_build ~= nil then
+    prerelease = parse_prerelease(prerelease_and_build)
+    build = parse_build(prerelease_and_build)
+  end
+
+  return {
+    major = major,
+    minor = minor,
+    patch = patch,
+    prerelease = prerelease,
+    build = build,
+  }
+end
+
+return M
```

Comment:
could do the setmetatable() stuff here, then we can name this file `version.lua`. Set the `__call` method to

```
   __call = function()
      return vim.fn.api_info().version
    end,
```

then we can delete the thing in executor.c